x/evm/keeper/grpc_query.go

```diff
@@ -507,10 +509,15 @@ func (k Keeper) TraceBlock(c context.Context, req *types.QueryTraceBlockRequest)
 	txConfig := statedb.NewEmptyTxConfig(common.BytesToHash(ctx.HeaderHash().Bytes()))
 	for i, tx := range req.Txs {
 		result := types.TxTraceResult{}
-		ethTx := tx.AsTransaction()
-		txConfig.TxHash = ethTx.Hash()
+		txConfig.TxHash = tx.AsTransaction().Hash()
+		coreMsg, err := tx.AsMessage(signer, cfg.BaseFee)
```

Comment:
we need to use `MsgEthereumTx.AsMessage` instead of `ethtypes.Transaction.AsMessage` so we can replace the sender